pkg/controllers/ipaddressallocation/ipaddressallocation_controller.go

```diff
@@ -142,42 +130,20 @@ func (r *IPAddressAllocationReconciler) handleUpdate(ctx context.Context, req ct
 }
 
 func (r *IPAddressAllocationReconciler) handleDeletion(ctx context.Context, req ctrl.Request, obj *v1alpha1.IPAddressAllocation) (ctrl.Result, error) {
-	if controllerutil.ContainsFinalizer(obj, servicecommon.IPAddressAllocationFinalizerName) {
-		metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerDeleteTotal, MetricResType)
-		if err := r.Service.DeleteIPAddressAllocation(obj); err != nil {
-			log.Error(err, "deletion failed, would retry exponentially", "IPAddressAllocation", req.NamespacedName)
-			deleteFail(r, ctx, obj, &err)
-			return resultRequeue, err
-		}
-		controllerutil.RemoveFinalizer(obj, servicecommon.IPAddressAllocationFinalizerName)
-		if err := r.Client.Update(ctx, obj); err != nil {
-			log.Error(err, "deletion failed, would retry exponentially", "IPAddressAllocation", req.NamespacedName)
-			deleteFail(r, ctx, obj, &err)
-			return resultRequeue, err
-		}
-		log.V(1).Info("removed finalizer on IPAddressAllocation CR", "IPAddressAllocation", req.NamespacedName)
-		deleteSuccess(r, ctx, obj)
-		log.Info("successfully deleted IPAddressAllocation CR and all subnets", "IPAddressAllocation", obj)
-	} else {
-		// only print a message because it's not a normal case
-		log.Info("IPAddressAllocation CR is being deleted but its finalizers cannot be recognized", "IPAddressAllocation", req.NamespacedName)
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerDeleteTotal, MetricResType)
+	if err := r.Service.DeleteIPAddressAllocation(obj); err != nil {
+		log.Error(err, "deletion failed, would retry exponentially", "IPAddressAllocation", req.NamespacedName)
+		deleteFail(r, ctx, obj, &err)
+		return resultRequeue, err
 	}
+	deleteSuccess(r, ctx, obj)
+	log.Info("successfully deleted IPAddressAllocation CR and all subnets", "IPAddressAllocation", obj)
 	return resultNormal, nil
 }
 
 func (r *IPAddressAllocationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.IPAddressAllocation{}).
-		WithEventFilter(predicate.Funcs{
-			UpdateFunc: func(e event.UpdateEvent) bool {
-				// Ignore updates to CR status in which case metadata.Generation does not change
-				return e.ObjectOld.GetGeneration() != e.ObjectNew.GetGeneration()
```

Comment:
Do we also expect to remove the filter on Update events?

Reply:
Only updating to the status subresource do not increment the generation. We could ignore it since we have no such scenario.